src/LodeRunner.API/src/Services/LRAPICacheService.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using LodeRunner.API.Data;
+using LodeRunner.API.Interfaces;
+using LodeRunner.API.Models;
+using LodeRunner.Services;
+using Microsoft.Azure.Documents;
+
+namespace LodeRunner.API.Services
+{
+    public class LRAPICacheService : ILRAPICacheService
+    {
```

Comment:
Is this intended to be the final form of the cache and cache service or is there another iteration?  When we chatted about it I was thinking we'd have generic methods that would internally create, put, and get from caches based on type.  So, LRAPICacheService.GetClientByClientStatusId(id) would end up calling something like cache.GetItem<ClientStatus>(id).

Let's chat about it.

Reply:
this does not include Caching Layer refactoring, there is an other branch out there that I created off "great merge" to minimize merge conflicts. On "great merge" branch I just reorganized Caching , it's intend is only to refractor Config an secrets